### ISSUE TITLE
Release backport of fixing `field_descriptors` to `field_specifiers` in `dataclass_transform` for Python 3.6?

### ISSUE THREAD

**Issue by bluenote10** [python/typing_extensions#74]
It looks like typing_extension versions <4.2 were contained a small bug related to [PEP 681](https://peps.python.org/pep-0681/#class-example): Within the  `dataclass_transform` decorator, the kwarg `field_specifiers` was erroneously called `field_descriptors`. This has recently been fixed for versions 4.2+.

The issue is that versions 4.2+ were not released for Python 3.6. This means that under Python 3.6 the code still looks like this:

```python
if hasattr(typing, 'dataclass_transform'):
    dataclass_transform = typing.dataclass_transform
else:
    def dataclass_transform(
        *,
        eq_default: bool = True,
        order_default: bool = False,
        kw_only_default: bool = False,
        field_descriptors: typing.Tuple[
            typing.Union[typing.Type[typing.Any], typing.Callable[..., typing.Any]],
            ...
        ] = (),
    ) -> typing.Callable[[T], T]:
        ...
```

This means that it is impossible to use `dataclass_transform` consistently for across Python 3.6 and later versions (code using the proper keyword `field_descriptors` crashes at runtime under Python 3.6). Furthermore PEP 681 requires these keywords to be specified statically, so generating the kwargs dynamically doesn't seem to work either.

Would it be possible to backport this fix and release an update of typing_extensions for Python 3.6?

**Comment by JelleZijlstra** [comment]
Python 3.6 has reached its end of life and we no longer support it.

**Comment by bluenote10** [comment]
I see. It's just a bit unfortunate that this one-liner hasn't been published before dropping Python 3.6 support... Will I now cave in and drop Python 3.6 support in my libraries as well :thinking:
